api/shared/compare/serializers.py

```diff
@@ -76,3 +90,172 @@ class FlagComparisonSerializer(serializers.Serializer):
+                    "head_number": line.number["head"],
+                    "base_coverage": line.coverage["base"],
+                    "head_coverage": line.coverage["head"],
+                    "content": content,
+                }
+            )
+        return lines
+
+
+class ImpactedFileSerializer(serializers.Serializer):
+    file_name = serializers.SerializerMethodField()
+    base_name = serializers.CharField()
+    head_name = serializers.CharField()
+    is_new_file = serializers.SerializerMethodField()
+    is_renamed_file = serializers.SerializerMethodField()
+    is_deleted_file = serializers.SerializerMethodField()
+    base_coverage = serializers.SerializerMethodField()
+    head_coverage = serializers.SerializerMethodField()
+    patch_coverage = serializers.SerializerMethodField()
+    change_coverage = serializers.SerializerMethodField()
+    misses_count = serializers.SerializerMethodField()
+    hashed_path = serializers.SerializerMethodField()
+    segments = serializers.SerializerMethodField()
+
+    def get_base_coverage(self, impacted_file: ImpactedFile) -> serializers.JSONField:
+        if impacted_file.base_coverage:
+            return dataclasses.asdict(impacted_file.base_coverage)
+
+    def get_head_coverage(self, impacted_file: ImpactedFile) -> serializers.JSONField:
+        if impacted_file.head_coverage:
+            return dataclasses.asdict(impacted_file.head_coverage)
+
+    def get_patch_coverage(self, impacted_file: ImpactedFile) -> serializers.JSONField:
+        if impacted_file.patch_coverage:
+            return dataclasses.asdict(impacted_file.patch_coverage)
+
+    def get_file_name(self, impacted_file: ImpactedFile) -> serializers.CharField:
+        return impacted_file.file_name
+
+    def get_is_new_file(self, impacted_file: ImpactedFile) -> serializers.BooleanField:
+        base_name = impacted_file.base_name
+        head_name = impacted_file.head_name
+        return base_name is None and head_name is not None
+
+    def get_is_renamed_file(
+        self, impacted_file: ImpactedFile
+    ) -> serializers.BooleanField:
+        base_name = impacted_file.base_name
+        head_name = impacted_file.head_name
+        return (
+            base_name is not None and head_name is not None and base_name != head_name
+        )
+
+    def get_is_deleted_file(
+        self, impacted_file: ImpactedFile
+    ) -> serializers.BooleanField:
+        base_name = impacted_file.base_name
+        head_name = impacted_file.head_name
+        return base_name is not None and head_name is None
+
+    def get_change_coverage(
+        self, impacted_file: ImpactedFile
+    ) -> serializers.FloatField:
+        return impacted_file.change_coverage
+
+    def get_misses_count(self, impacted_file: ImpactedFile) -> serializers.IntegerField:
+        return impacted_file.misses_count
+
+    def get_hashed_path(self, impacted_file: ImpactedFile) -> serializers.CharField:
```

Comment:
This might be unnecessary - I think this was for some niche use case in Gazebo.  Not sure if it would be generally useful.